### PR TITLE
Add virtual machines to supporting services

### PR DESCRIPTION
**Pull request by mateo762**
This pull request includes changes to the `Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/main.bicep` file and related files to support the addition of virtual machines and a container registry. The changes include the addition of new parameters, types, and modules for the virtual machines and container registry, as well as updates to the naming conventions.

Additions to parameters and types:

* [`Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/main.bicep`](diffhunk://#diff-3e31e0003f9854439baf1661a4493575a18bf82a7fd135b7be46fcc4eced57a9R64-R66): Added parameters for the jump box subnet, virtual machine size, admin username and password, NIC configurations, OS disk configuration, and container registry SKU. [[1]](diffhunk://#diff-3e31e0003f9854439baf1661a4493575a18bf82a7fd135b7be46fcc4eced57a9R64-R66) [[2]](diffhunk://#diff-3e31e0003f9854439baf1661a4493575a18bf82a7fd135b7be46fcc4eced57a9R102-R173)
* [`Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/modules/virtual-machine/types.bicep`](diffhunk://#diff-37e79a5084b1afe475fa4dc9bab82c88bac88d07a5ac4917e8f2f79f2f7d56b5R1-R56): Added types for image reference, NIC configuration, OS disk, IP configuration, and managed disk.
* [`Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/modules/container-registry/types.bicep`](diffhunk://#diff-93b60d81f19b7212c7537cdcd1540e37c0a88641c41e76b7f64815ee82dc0ef6R1-R3): Added type for container registry SKU.

Additions to modules:

* [`Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/main.bicep`](diffhunk://#diff-3e31e0003f9854439baf1661a4493575a18bf82a7fd135b7be46fcc4eced57a9R234-R280): Added modules for Windows and Linux virtual machines and container registry.

Updates to naming conventions:

* [`Scenarios/Secure-Baseline/bicepWithAVM/commonModules/naming/abbreviations.json`](diffhunk://#diff-6c2ccc5fd3235c6844cdf42018fe0b01da60b1a2b60a58c2d8c416935181ee78R22): Added abbreviation for container registry.
* [`Scenarios/Secure-Baseline/bicepWithAVM/commonModules/naming/types.bicep`](diffhunk://#diff-252039a03259cffba89bf870b7d332324059f9c4c6b7898f98d67db5d4dfaec8L3-R3): Added container registry to resource type type.

Imports and other changes:

* [`Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/main.bicep`](diffhunk://#diff-3e31e0003f9854439baf1661a4493575a18bf82a7fd135b7be46fcc4eced57a9R8-R10): Imported types from the virtual machine and container registry modules.

### CODE REVIEW

File: Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/main.bicep [hunk 1]
```diff
@@ -153,3 +231,50 @@ module keyVault 'br/public:avm/res/key-vault/vault:0.6.2' = {
...
+    osType: 'Linux'
+    vmSize: vmSize
+    location: location
+    zone: 0
+    enableTelemetry: enableAvmTelemetry
+  }
+}
+
+/* -------------------------------- Container Registry ------------------------------------------ */
+module registry 'br/public:avm/res/container-registry/registry:0.3.1' = {
```
Comment (pmalarme) [inline Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/main.bicep:273]: Make container registry private

File: Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/main.bicep [hunk 2]
```diff
@@ -93,6 +99,78 @@ param secrets array = []
 @description('The name of the private endpoint for the key vault. Defaults to the naming convention `<abbreviation-private-endpoint>-<key-vault-name>`.')
 param keyVaultPrivateEndpointName string = getResourceNameFromParentResourceName('privateEndpoint', keyVaultName, null, hash)
 
+/* ------------------------------- Virtual Machine ------------------------------- */
+// Common parameters for both VMs
```
Comment (pmalarme) [inline Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/main.bicep:103]: Separate the parameters of the VM

File: Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/main.bicep [hunk 3]
```diff
@@ -5,6 +5,9 @@ targetScope = 'resourceGroup'
 /* -------------------------------------------------------------------------- */
 
 import { skuType as keyVaultSkuType } from './modules/key-vault/types.bicep'
+import { imageReferenceType, nicConfigurationType, osDiskType } from './modules/virtual-machine/types.bicep'
+import { containerRegistrySkuType } from './modules/container-registry/types.bicep'
```
Comment (pmalarme) [inline Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/main.bicep:9]: use skuType and an alias like for the key vault

**Review by pmalarme — CHANGES_REQUESTED** [review]

File: Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/main.bicep (hunk 3 above)
Comment (mateo762) [reply Scenarios/Secure-Baseline/bicepWithAVM/03-Supporting-Services/main.bicep:9]: The `skuType` for Key Vault and for Container Registry differ in values. 

Key Vault takes the values `premium | standard | null`.
Container Registry takes the values `Premium | Standard | Basic | null`.

I will create a separate `skuType` inside `modules/container-registry/types.bicep` and import it using an alias.

**Review by pmalarme — APPROVED** [review]

**Pull request merged**